### PR TITLE
(SIMP-7465) Update to handle chronyd default

### PR DESCRIPTION
**Pull request by jeannegreulich**
- Add item in network task list to create a hiera link from
  chronyd::servers to simp_options::ntp::servers.
- Update `simp config` to check for chrony settings
  when determining the currently set os value.

SIMP-7465 #close
SIMP-8966 #close

### CODE REVIEW

File: spec/lib/simp/cli/commands/files/simp_conf_setting_values_poss_scenario.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+chronyd::servers: "%{hiera('simp_options::ntp::servers')}"
```
Comment (op-ct) [inline spec/lib/simp/cli/commands/files/simp_conf_setting_values_poss_scenario.yaml:2]:
```suggestion
chronyd::servers: "%{alias('simp_options::ntp::servers')}"
```

File: spec/lib/simp/cli/commands/files/simp_conf_setting_values_simp_lite_scenario.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
+chronyd::servers: "%{hiera('simp_options::ntp::servers')}"
```
Comment (op-ct) [inline spec/lib/simp/cli/commands/files/simp_conf_setting_values_simp_lite_scenario.yaml:2]:
```suggestion
chronyd::servers: "%{alias('simp_options::ntp::servers')}"
```

File: spec/lib/simp/cli/commands/files/simp_conf_with_overrides.yaml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ---
+chronyd::servers: "%{hiera('simp_options::ntp::servers')}"
```
Comment (op-ct) [inline spec/lib/simp/cli/commands/files/simp_conf_with_overrides.yaml:2]:
```suggestion
chronyd::servers: "%{alias('simp_options::ntp::servers')}"
```

File: spec/lib/simp/cli/commands/files/prev_simp_conf.yaml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 ---
+chronyd::servers: "%{hiera('simp_options::ntp::servers')}"
```
Comment (op-ct) [inline spec/lib/simp/cli/commands/files/prev_simp_conf.yaml:2]:
```suggestion
chronyd::servers: "%{alias('simp_options::ntp::servers')}"
```

File: spec/lib/simp/cli/config/items/data/simp_options_ntp_servers_spec.rb [hunk 5]
```diff
@@ -44,26 +44,70 @@
...
+      allow(Simp::Cli::Utils).to receive(:systemctl_running?).with('ntpd').and_return(false)
+      expect( @ci.get_os_value(@chrony_local_servers,@ntp_remote_servers)).to eq []
+    end
+
+    it 'returns empty array when local servers are in the ntps files' do
+      allow(Simp::Cli::Utils).to receive(:systemctl_running?).with('chronyd').and_return(false)
+      allow(Simp::Cli::Utils).to receive(:systemctl_running?).with('ntpd').and_return(true)
+      expect( @ci.get_os_value('/not/there',@ntp_local_servers)).to eq []
+    end
+    it 'returns the ntp servers when no services are running and not chrony file exists' do
```
Comment (op-ct) [inline spec/lib/simp/cli/config/items/data/simp_options_ntp_servers_spec.rb:103]:
```suggestion
    it 'returns the ntp servers when no services are running and no chrony file exists' do
```

**Review by op-ct — CHANGES_REQUESTED** [review]

**Comment by lnemsick-simp** [comment]
simp_options_ntp_servers.rb needs to change is `@key` from `simp_options::ntpd::servers` to `simp_options::ntp::servers`.

**Comment by lnemsick-simp** [comment]
I'll fix the `simp_options::ntdp::servers` references throughout the code and tests and update the changelog entry accordingly

File: lib/simp/cli/config/items/data/chronyd_link_to_ntpservers.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+require_relative '../list_item'
+require_relative '../../utils'
+require_relative 'cli_network_gateway'
+
+module Simp; end
+class Simp::Cli; end
+module Simp::Cli::Config
+  class Item::ChronydNTPServersDefault < Item
```
Comment (lnemsick-simp) [inline lib/simp/cli/config/items/data/chronyd_link_to_ntpservers.rb:8]: Class should be named for hieradata setting and name of the file should match the class name.  I'll fix.

File: lib/simp/cli/config/items/data/chronyd_link_to_ntpservers.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    def initialize(puppet_env_info = DEFAULT_PUPPET_ENV_INFO)
+      super(puppet_env_info)
+      @key          = 'chronyd::servers'
```
Comment (lnemsick-simp) [inline lib/simp/cli/config/items/data/chronyd_link_to_ntpservers.rb:11]: hieradata key should be chrony::servers.  I'll fix

**Review by lnemsick-simp — APPROVED** [review]

**Comment by lnemsick-simp** [comment]
Acceptance test failures are due to GitLab runner configuration, not the code changes.

**Pull request merged**